torch/distributed/_tensor/debug/visualize_sharding.py

```diff
@@ -135,6 +135,9 @@ def visualize_sharding(dtensor, prompt=""):
     Visualizes sharding in 1D-2D dtensors
     Requires tabulate, install with `pip install tabulate`
     """
+    if (dtensor.numel() == 0):  # we do not print for empty tensors
```

Comment:
hmmm is this right? this only print empty DTensor but not the case where it's a DTensor that some ranks have data but some ranks have empty tensor?

Reply:
yeah, you're right. 2 things to address:
1) if the DTensor is empty, visualize_sharding will throw exception when calculating the offset. So I skip if this is the case in code in this PR.
2) if the DTensor is placed on a submesh, visualize_sharding will throw exception when calling `get_local_rank()` because non-participating ranks have empty `_dim_group_infos`. So I skip on all non-participating ranks. This change is in #121382